code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -49,6 +49,13 @@
 	mechanics_text = "Plant a weed node (purple sac) on your tile."
 	keybind_signal = COMSIG_XENOABILITY_DROP_WEEDS
 
+/datum/action/xeno_action/plant_weeds/can_use_action(silent, override_flags)
+	. = ..()
+	if(locate(/obj/effect/alien/resin/trap) in get_turf(owner))
```

Comment:
This will be called all the time, constantly locating for traps under the mob, even if it's not trying to use the ability. It would be better if this was done on demand, in `action_activate()`, no?

Also, shouldn't this be on `check_alien_construction()`?

Reply:
check_alien_construction fails on a lot of things weeds shouldnt fail on.

can_use_action should in theory be on demand, sadly it looks like theres a proc chain from life that will trigger it.

in terms of xeno actions, can_use_* is meant to check if the action can be done, and action_activate/use_ability do it without (much) checks

Reply:
So keep this as it is, and plan to remove the call from `Life()`?